test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs

```diff
@@ -321,5 +319,22 @@ public class ActionResultTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
             Assert.Equal("{\"SampleInt\":10,\"SampleString\":\"Foo\"}", await response.Content.ReadAsStringAsync());
         }
+
+        [Fact]
+        public async Task HttpNotFoundObjectResult_WithDisposableObject()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName);
+            var client = server.CreateClient();
+
+            var request = new HttpRequestMessage(
+                HttpMethod.Post,
+                "http://localhost/ActionResultsVerification/GetNotFoundObjectResultWithDisposableObject");
+
+            // Act & Assert
+            var exception = await Assert.ThrowsAsync<Exception>(
```

Comment:
hmm..wondering if this test is passing because we are running it 'in-memory'? I say this because the dispose on the resource is called 'after' the response has been written out and so any exception happening in disposed method should not be sent back to the client...i suspect this test would break if the service is hosted on a real server...

Some options you can consider are: 
- have  a static variable which increments the count when dispose is called. so one regular request and one more request to get the current count of the dispose calls and verify.
- use logger to log messages when dispose is called and verify the log messages. Here also i think you would need 2 requests to verify.